inductiva/utils/templates.py

```diff
@@ -1,83 +1,74 @@
 """Utils related to template files."""
 
 import os
-import io
+import pathlib
 import glob
-from typing import Dict, List, Union
-from pathlib import Path
+import shutil
 
-from jinja2 import Environment, FileSystemLoader
+import jinja2
 
 from inductiva.utils.files import find_path_to_package
 
 TEMPLATES_PATH = find_path_to_package("templates")
 
 
-def replace_params(
-    template_path: str,
-    params: Dict,
-    output_file: Union[str, io.StringIO],
-    remove_template: bool = False,
-) -> None:
-    """Replaces parameters in a template file."""
+def render_from_string(template_string, **render_args):
+    """Render commands with the given arguments."""
 
-    template_path = Path(template_path)
+    template_env = jinja2.Environment(loader=jinja2.BaseLoader())
+    template = template_env.from_string(template_string)
+    string = template.render(**render_args)
 
-    template_dir = template_path.parent
-    template_filename = template_path.name
+    return string
 
-    environment = Environment(loader=FileSystemLoader(template_dir))
-    template = environment.get_template(template_filename)
-    stream = template.stream(**params)
-    stream.dump(output_file)
+
+def render_file(source_file, target_file, remove_template=False, **render_args):
+
+    source_path = pathlib.Path(source_file)
+
+    source_dir = source_path.parent
+    source_file = source_path.name
+
+    environment = jinja2.Environment(loader=jinja2.FileSystemLoader(source_dir))
+    template = environment.get_template(source_file)
+    stream = template.stream(**render_args)
+    stream.dump(target_file)
```

Comment:
is the stream properly disposed after this point?

Reply:
I don't find anything about the stream needing to be closed. But I am going to go over it.

Reply:
If users want to a stream dump again, it doesn't provide anything.

By the documentation: http://mitsuhiko.pocoo.org/jinja2docs/Jinja2.pdf it doesn't mention anyway to close the stream - presumably it is automatic.

Reply:
Thanks.
Yes, the stream is closed at the end (see [here](https://github.com/pallets/jinja/blob/d594969d722ceb4e8f3da8861befc9c0ac87ae1b/src/jinja2/environment.py#L1626)).